src/GraphQL/Resolver.hs

```diff
@@ -251,14 +252,14 @@ instance forall ks t f m.
   , HasAnnotatedInputType t
   , Monad m
   ) => BuildFieldResolver m (API.Argument ks t :> f) where
-  buildFieldResolver handler selection@(AST.SelectionField (AST.Field _ _ arguments _ _)) = do
+  buildFieldResolver handler field = do
     argument <- first SchemaError (API.getArgumentDefinition @(API.Argument ks t))
     let argName = getName argument
+    let arguments = getArguments field
```

Comment:
As I might have mentioned somewhere, I'm working on variable validation atm, so this change could go in there or in this PR.